### PR TITLE
Proxy Farm ExecuteOnDest + nft transfer calls refactor

### PR DESCRIPTION
**Pull request by catalinnnn**

### CODE REVIEW

File: distribution/sc-proxy-dex/src/proxy_farm.rs [hunk 1]
```diff
@@ -348,15 +318,94 @@ pub trait ProxyFarmModule: proxy_common::ProxyCommonModule + proxy_pair::ProxyPa
...
+            accented_funds_len == 1 || accented_funds_len == 2,
+            "Bad received funds len"
+        );
+
+        let asset_token_id = self.asset_token_id().get();
+        let locked_asset_token_id = self.locked_asset_token_id().get();
+        let mut farming_token = Option::<FftTokenAmountPair<Self::BigUint>>::None;
+        let mut reward_token = Option::<GenericEsdtAmountPair<Self::BigUint>>::None;
+
+        for (token_id, token_nonce) in self.current_tx_accepted_funds().keys() {
```
Comment (sasurobert) [inline distribution/sc-proxy-dex/src/proxy_farm.rs:347]: wouldn't be better to use the validate function ? it looks relatively the same.

Comment (sasurobert) [reply distribution/sc-proxy-dex/src/proxy_farm.rs:347]: like lines 342-385 could be taken out into a validate function

File: distribution/sc-proxy-dex/src/proxy_farm.rs [hunk 2]
```diff
@@ -365,15 +414,72 @@ pub trait ProxyFarmModule: proxy_common::ProxyCommonModule + proxy_pair::ProxyPa
...
+            accented_funds_len == 1 || accented_funds_len == 2,
+            "Bad received funds len"
+        );
+
+        let asset_token_id = self.asset_token_id().get();
+        let locked_asset_token_id = self.locked_asset_token_id().get();
+        let mut farm_token = Option::<GenericEsdtAmountPair<Self::BigUint>>::None;
+        let mut reward_token = Option::<GenericEsdtAmountPair<Self::BigUint>>::None;
+
+        for (token_id, token_nonce) in self.current_tx_accepted_funds().keys() {
```
Comment (sasurobert) [inline distribution/sc-proxy-dex/src/proxy_farm.rs:443]: somewhat duplicated code with previous validation.

File: distribution/sc-proxy-dex/src/proxy_farm.rs [hunk 3]
```diff
@@ -137,25 +137,10 @@ pub trait ProxyFarmModule: proxy_common::ProxyCommonModule + proxy_pair::ProxyPa
 
         self.reset_received_funds_on_current_tx();
         let farm_result = self
-            .actual_exit_farm(&farm_address, &farm_token_id, farm_token_nonce, &amount)
+            .actual_exit_farm(&farm_address, &farm_token_id, farm_token_nonce, &amount)?
             .into_tuple();
         let farming_token_returned = farm_result.0;
         let reward_token_returned = farm_result.1;
```
Comment (sasurobert) [inline distribution/sc-proxy-dex/src/proxy_farm.rs:143]: why did you delete the validate_received_funds_chunk - I think it should be kept here.

**Review by sasurobert — CHANGES_REQUESTED** [review]

File: distribution/sc-proxy-dex/src/proxy_farm.rs (hunk 1 above)
Comment (catalinnnn) [reply distribution/sc-proxy-dex/src/proxy_farm.rs:410]: done

File: distribution/sc-proxy-dex/src/proxy_farm.rs (hunk 3 above)
Comment (catalinnnn) [reply distribution/sc-proxy-dex/src/proxy_farm.rs:147]: done

File: distribution/sc-proxy-dex/src/proxy_farm.rs (hunk 2 above)
Comment (catalinnnn) [reply distribution/sc-proxy-dex/src/proxy_farm.rs:512]: both do pretty much the same thing, yes, but the return types differ and also the checks differ

**Review by sasurobert — APPROVED** [review]
